crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -390,6 +382,130 @@ System site-packages will not be used for module resolution.",
     }
 }
 
+/// A parser for `pyvenv.cfg` files: metadata files for virtual environments.
+///
+/// Note that a `pyvenv.cfg` file *looks* like a `.ini` file, but actually isn't valid `.ini` syntax!
+///
+/// See also: <https://snarky.ca/how-virtual-environments-work/>
+#[derive(Debug)]
+struct PyvenvCfgParser<'s> {
+    cursor: Cursor<'s>,
+    line_number: NonZeroUsize,
+    data: RawPyvenvCfg<'s>,
+}
```

Comment:
The switch to a proper parser here is motivated by wanting to capture the `TextRange` of the Python version in the `pyvenv.cfg` file so that we can display it in diagnostics. Screenshot of what this looks like on the command line:

![image](https://github.com/user-attachments/assets/0aa68490-b9cc-4e85-a58c-64dff1f0e925)